src/c_api/c_api.cc

```diff
@@ -1924,6 +1967,194 @@ int MXDataIterGetPadNum(DataIterHandle handle, int *pad) {
   API_END();
 }
 
+int MXListDatasets(uint32_t *out_size,
+                             DatasetCreator **out_array) {
+  API_BEGIN();
+  auto &vec = dmlc::Registry<DatasetReg>::List();
+  *out_size = static_cast<uint32_t>(vec.size());
+  *out_array = (DatasetCreator*)(dmlc::BeginPtr(vec));  //  NOLINT(*)
+  API_END();
+}
+
+int MXDatasetCreateDataset(DatasetCreator handle,
+                           uint32_t num_param,
+                           const char **keys,
+                           const char **vals,
+                           DatasetHandle *out) {
+  Dataset *dataset = nullptr;
+  API_BEGIN();
+  DatasetReg *e = static_cast<DatasetReg *>(handle);
+  dataset = e->body();
```

Comment:
Need to pass kwargs here after changing registry func type.

Reply:
okay, I got your point, will change to constructor.

Reply:
Thanks. For reference you can check https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-complete why this is a good practice.